app-deployment-manager/test/deployment_list/deployment_list_test.go

```diff
@@ -0,0 +1,82 @@
+// SPDX-FileCopyrightText: (C) 2025-present Intel Corporation
+//
+// SPDX-License-Identifier: Apache-2.0
+
+package deployment_list
+
+import (
+	"net/http"
+
+	"github.com/open-edge-platform/app-orch-deployment/app-deployment-manager/api/nbi/v2/pkg/restClient"
+	deploymentutils "github.com/open-edge-platform/app-orch-deployment/test-common-utils/pkg/deployment"
+)
+
+func ptr[T any](v T) *T {
+	return &v
+}
+
+func (s *TestSuite) TestListDeploymentsWithPagination() {
+	s.T().Parallel()
+	testName := "ListDeploymentsWithPagination"
+	for _, app := range []string{deploymentutils.AppWordpress, deploymentutils.AppNginx} {
+		deploymentReq := deploymentutils.StartDeploymentRequest{
+			AdmClient:         s.AdmClient,
+			DpPackageName:     app,
+			DeploymentType:    deploymentutils.DeploymentTypeTargeted,
+			DeploymentTimeout: deploymentutils.DeploymentTimeout,
+			DeleteTimeout:     deploymentutils.DeleteTimeout,
+			TestName:          testName,
+		}
+		_, code, err := deploymentutils.StartDeployment(deploymentReq)
+		s.Equal(http.StatusOK, code)
+		s.NoError(err, "Failed to create '"+app+"-"+deploymentutils.DeploymentTypeTargeted+"' deployment")
+	}
+
+	deps, code, err := deploymentutils.DeploymentsListWithParams(s.AdmClient, &restClient.DeploymentServiceListDeploymentsParams{
```

Comment:
Can we extend this test a little bit or add new tests to test pageSize and offset in out of range or negative values? if it makes sense we should add new tests for invalid use cases if we don't have one.

Reply:
done